torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -200,11 +230,14 @@ def _share_state_and_init_handle_attrs(
     root_state: _FSDPState,
     root_module: nn.Module,
 ) -> None:
-    """
-    Shares data structure state from the ``root_state`` to all FSDP states in
-    ``root_module`` 's module tree, and initializes handle attributes. These
-    are done together to require a single loop over the states.
-    """
+    """Shares data structure state from the ``root_state`` to all FSDP states in
+    ``root_module`` 's module tree, and initializes handle attributes. These are
+    done together to require a single loop over the states."""
+    from torch.distributed.fsdp._runtime_utils import (
+        HOMOGENEOUS_ATTR_NAMES, _init_device_mesh,
+        _validate_and_get_hybrid_shard_state)
+    from torch.distributed.utils import _p_assert
```

Comment:
This is already in the global namescope of the python file? Any reason to reimport it?

Reply:
oop just copy-pasted from my patch. Will remove this